pre_commit/main.py

```diff
@@ -123,6 +123,18 @@ def _add_run_options(parser: argparse.ArgumentParser) -> None:
         '--files', nargs='*', default=[],
         help='Specific filenames to run hooks on.',
     )
+    parser.add_argument(
+        '--prev-head-ref',
+        help='The ref of the previous HEAD.',
+    )
+    parser.add_argument(
+        '--new-head-ref',
+        help='The ref of the new HEAD (which may or may not have changed).',
+    )
```

Comment:
I think we could potentially reuse `--source` / `--origin` for these arguments?

Reply:
I can do that if you want to keep the arg count down. However, I don't think that source/origin maps directly to new-head-ref/prev-head-ref so it might create some confusion. I'll leave that decision up to you.

Reply:
`source` / `origin` is a "from ref" -> "to ref" relation, so I think it fits well enough

Reply:
Ok cool, I will do that.